Marlin/Marlin_main.cpp

```diff
@@ -4907,7 +4998,136 @@ inline void gcode_M226() {
   }
 #endif // E_REGULATOR
 
+#if ENABLED(AUTO_BED_LEVELING_FEATURE) && ENABLED(EXT_ADC)
+  /*
+  * M237 - Custom, more precise auto bed leveling
+  */
+  inline void gcode_M237() {
+    if (!axis_known_position[X_AXIS] || !axis_known_position[Y_AXIS]) {
+      LCD_MESSAGEPGM(MSG_POSITION_UNKNOWN);
+      SERIAL_ECHO_START;
+      SERIAL_ECHOLNPGM(MSG_POSITION_UNKNOWN);
+      return;
+    }
+  
+    int verbose_level = code_seen('V') || code_seen('v') ? code_value_short() : 1;
+    if (verbose_level < 0 || verbose_level > 4) {
+      SERIAL_ECHOLNPGM("?(V)erbose Level is implausible (0-4).");
+      return;
+    }
+  
+    bool dryrun = code_seen('D') || code_seen('d');
+    st_synchronize();
+  
+    if (!dryrun) {
+      plan_bed_level_matrix.set_to_identity();
+      #ifdef DELTA
+        reset_bed_level();
+      #else
+        vector_3 uncorrected_position = plan_get_position();
+        current_position[X_AXIS] = uncorrected_position.x;
+        current_position[Y_AXIS] = uncorrected_position.y;
+        current_position[Z_AXIS] = uncorrected_position.z;
+        sync_plan_position();
+      #endif // !DELTA
+    }
+  
+    setup_for_endstop_move();
+    feedrate = homing_feedrate[Z_AXIS];
+  
+    // Need to make sure we're at 1.00 on the Z-axis
+    destination[Z_AXIS] = BED_LEVEL_PROBE_Z;
+    prepare_move();
+  
+    float levelProbe_1 = bed_level_probe_pt(ABL_PROBE_PT_1_X - X_PROBE_OFFSET_FROM_EXTRUDER, ABL_PROBE_PT_1_Y - Y_PROBE_OFFSET_FROM_EXTRUDER, current_position[Z_AXIS], verbose_level),
+          levelProbe_2 = bed_level_probe_pt(ABL_PROBE_PT_2_X - X_PROBE_OFFSET_FROM_EXTRUDER, ABL_PROBE_PT_2_Y - Y_PROBE_OFFSET_FROM_EXTRUDER, current_position[Z_AXIS], verbose_level),
+          levelProbe_3 = bed_level_probe_pt(ABL_PROBE_PT_3_X - X_PROBE_OFFSET_FROM_EXTRUDER, ABL_PROBE_PT_3_Y - Y_PROBE_OFFSET_FROM_EXTRUDER, current_position[Z_AXIS], verbose_level);
+    
+    levelProbe_1 = (levelProbe_1 - 5000)/1000;
+    levelProbe_2 = (levelProbe_2 - 5000)/1000;
+    levelProbe_3 = (levelProbe_3 - 5000)/1000;
```

Comment:
I agree with @riningerr, they should be constants defined by the distance sensor code